### PR TITLE
New Editing Functionality Architecture

### LINKED ISSUE #31: Edit Entries Capacity
# Summary
Allow for the editing of Headwords

### PR DESCRIPTION
**Pull request by Shawk031**
# Summary
# Closes #31 Adds the architecture a proposed Entry Editing solution

## Notes
I think using a popup is our best shot at doing Entry Editing. I considered a lot of alternatives before stumbling onto this one, such as:
1. second edit form
2. passing down an "editable" prop and changing the form based on that (we'll still probably have "editable" as context info, so that we can hide the popup when a user isn't verified)
3. Replacing current Entry with a form that is only mutable if you're in Edit mode

I don't remember what else. But there are a lot of subtle problems to navigate here. The `Popover` element helps us bypass practically all of those additional problems

## Testing
Basically: 
1. Go to any word. 
4. Then, press the red book icon next to the Headword (We'll need to change the icon later).
5. Enter the new Headword you want to replace the current headword with
6. Verify that the headword did indeed change for that entry in the database (not sure if you know about prisma studio or not, but `npx prisma studio` is really really useful)

### CODE REVIEW

File: app/models/entry.server.ts [hunk 1]
```diff
@@ -142,3 +143,32 @@ export function getEntriesByBasicTextSearch(
       LOWER(headword) LIKE LOWER(${searchWildcard}))  
     ORDER BY headword ASC LIMIT ${take} OFFSET ${skip}`
 }
+
+export async function updateRecordByAttributeAndType(
+  type: number, // probably change this to enum
```
Comment (hangler) [inline app/models/entry.server.ts:148]: Yes, this will need to change to the type of the enum.

File: app/routes/entries.tsx [hunk 2]
```diff
@@ -1,7 +1,25 @@
...
 import Header from "~/components/elements/Header"
 import Main from "~/components/elements/Main"
 import Nav from "~/components/elements/Nav"
+import { updateRecordByAttributeAndType } from "~/models/entry.server"
+import {
+  getNumberFromFormInput,
+  getStringFromFormInput,
+} from "~/utils/generalUtils"
+
+export async function action({ request }: ActionArgs) {
```
Comment (hangler) [inline app/routes/entries.tsx:12]: Move to entries/headword route so user stays on the same page.

**Review by hangler — CHANGES_REQUESTED** [review]
Minor changes as we discussed.

File: app/models/entry.server.ts (hunk 1 above)
Comment (Shawk031) [reply app/models/entry.server.ts]: done!

File: app/routes/entries.tsx (hunk 2 above)
Comment (Shawk031) [reply app/routes/entries.tsx]: also done-- this was a bit stranger, cause there were problems with the redirect etc. unfortunately. So there's a bit of a workaround in the code
